src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/MSBuildProject.cs

```diff
@@ -90,13 +90,43 @@ public MBuild.Project Project
                 {
                     return Context.ProjectCollection.LoadProject(FileInfo.FullName);
                 }
-                catch (InvalidProjectFileException ex)
+                catch (InvalidProjectFileException e)
                 {
-                    throw new UpgradeException(LocalizedStrings.InvalidProjectError, ex);
+                    // Delete PhotoLab.csproj.nuget.g.* files and retry. A change in platform version may cause load to fail for UWP / WinUI apps
+                    if (!DeleteNugetGeneratedFiles())
+                    {
+                        throw new UpgradeException(LocalizedStrings.InvalidProjectError, e);
+                    }
+
+                    try
+                    {
+                        return Context.ProjectCollection.LoadProject(FileInfo.FullName);
+                    }
+                    catch (InvalidProjectFileException ex)
+                    {
+                        throw new UpgradeException(LocalizedStrings.InvalidProjectError, ex);
+                    }
                 }
             }
         }
 
+        private bool DeleteNugetGeneratedFiles()
+        {
+            if (FileInfo.Directory is null)
+            {
+                return false;
+            }
+
+            var filesToDelete = Directory.GetFiles(FileInfo.Directory.FullName, "PhotoLab.csproj.nuget.g.*", SearchOption.AllDirectories);
+            foreach (var filePath in filesToDelete)
+            {
+                File.Delete(filePath);
+                return true;
```

Comment:
This `return` makes it so only the first file in the list gets deleted. Is that accurate?